### PR TITLE
HTS-1399: Revisit RollNumber validation

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk.gov.hmrc.helptosaveproxy/models/NSIPayload.scala [hunk 1]
```diff
@@ -113,7 +113,7 @@ object NSIPayload {
       BankDetails(
         bankDetails.sortCode.cleanupSpecialCharacters.removeAllSpaces.filterNot(allowedSortCodeSeparators.contains).grouped(2).mkString("-"),
         bankDetails.accountNumber.cleanupSpecialCharacters.removeAllSpaces,
-        bankDetails.rollNumber.map(_.cleanupSpecialCharacters.removeAllSpaces),
+        bankDetails.rollNumber,
```
Comment (AndyHWChung) [inline app/uk.gov.hmrc.helptosaveproxy/models/NSIPayload.scala]: I think we'd still want to clean up special characters. That does this:
```scala
def cleanupSpecialCharacters: String = s.replaceAll("\t|\n|\r", " ").trim.replaceAll("\\s{2,}", " ")
```
which trims tabs, new lines, carriage returns and trims trailing and leading spaces

Comment (sureshhmrc) [reply app/uk.gov.hmrc.helptosaveproxy/models/NSIPayload.scala]: the frontend already takes care of those as it doesn't allow the spaces and `\t|\n|\r` ?

Comment (AndyHWChung) [reply app/uk.gov.hmrc.helptosaveproxy/models/NSIPayload.scala]: that's true - but the API service won't have that logic in place

Comment (sureshhmrc) [reply app/uk.gov.hmrc.helptosaveproxy/models/NSIPayload.scala]: yeah, i added a comment to the other PR if we need to add the validation in the api layer as well ?
is there any reason why we are not validating the bank details in the api layer ?

Comment (AndyHWChung) [reply app/uk.gov.hmrc.helptosaveproxy/models/NSIPayload.scala]: for the API journey there's no real incentive to have the checks up front as we'd get the same result if the request made it's way to the proxy. In the FE the reason we have the logic there as well is that there we can stop people on the journey if their bank details are wrong before they create an account, but we can't do that sort of thing in the API journey

Comment (sureshhmrc) [reply app/uk.gov.hmrc.helptosaveproxy/models/NSIPayload.scala]: ok, i added the logic back

**Pull request merged**